extensions/markdown-language-features/src/features/workspaceSymbolProvider.ts

```diff
@@ -26,10 +26,26 @@ class VSCodeWorkspaceMarkdownDocumentProvider extends Disposable implements Work
 
 	private _watcher: vscode.FileSystemWatcher | undefined;
 
-	async getAllMarkdownDocuments() {
+	/**
+	 * Reads and parses all .md documents in the workspace.
+	 * Files are processed in batches, to keep the number of open files small.
+	 *
+	 * @returns Array of processed .md files.
+	 */
+	async getAllMarkdownDocuments(): Promise<SkinnyTextDocument[]> {
+		const maxConcurrent = 20;
+		const docList: SkinnyTextDocument[] = [];
 		const resources = await vscode.workspace.findFiles('**/*.md', '**/node_modules/**');
-		const docs = await Promise.all(resources.map(doc => this.getMarkdownDocument(doc)));
-		return docs.filter(doc => !!doc) as SkinnyTextDocument[];
+
+		// add 1 in case of any remaining files,
+		// ex (10 % 3) = 3, needs 1 more loop to complete
+		for (let i = 0; i < resources.length % maxConcurrent + 1; i++) {
```

Comment:
Instead of using `%`, could you do something like: `for (let i = 0; I < resource.length; i += maxConcurrent)`